### PR TITLE
Python: bugfix: min relevance score calculation filtering on mongodbatlas semantic search

### PR DESCRIPTION
**Pull request by Jibola**
### Motivation and Context
#4433 

### Description

`$match` operation queries need a different structure than provided. As a result users receive `OperationFailure`s when trying to run queries. This introduces a fix to provide to correct query for filtering results by the min_relevance_score. 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by Jibola** [comment]
cc: @ShaneHarvey, @blink1073 , @caseyclements, @NoahStapp

**Review by ShaneHarvey — APPROVED** [review]
Is there a changelog or any other docs to update?

**Comment by Jibola** [comment]
> Is there a changelog or any other docs to update?

None that I've seen we need to contribute to?

File: python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ async def get_nearest_matches(
         pipeline.append({"$set": {"score": {"$meta": "vectorSearchScore"}}})
 
         if min_relevance_score:
```
Comment (ShaneHarvey) [inline python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:268]: Can score be negative? If so, this should be `min_relevance_score is not None` to handle the zero case?

Comment (Jibola) [inline python/semantic_kernel/connectors/memory/mongodb_atlas/mongodb_atlas_memory_store.py:268]:
```suggestion
        if min_relevance_score is not None:
```

**Comment by Jibola** [comment]
@markwallace-microsoft would you be able to take a look at this? It's a high priority issue for us.

**Review by moonbox3 — APPROVED** [review]

**Review by juliomenendez — APPROVED** [review]

**Comment by moonbox3** [comment]
@Jibola all good to go!

**Pull request merged**